jupyter_client/discovery.py

```diff
@@ -0,0 +1,80 @@
+from .kernelspec import KernelSpecManager
+from .manager import KernelManager
+
+
+class KernelSpecFinder(object):
+    """Find kernels from installed kernelspec directories.
+    """
+    id = 'spec'
+
+    def __init__(self):
+        self.ksm = KernelSpecManager()
+
+    def find_kernels(self):
+        for name, resdir in self.ksm.find_kernel_specs().items():
+            spec = self.ksm._get_kernel_spec_by_name(name, resdir)
+            yield name, {
+                # TODO: get full language info
+                'language': {'name': spec.language},
+                'display_name': spec.display_name,
+                'argv': spec.argv,
```

Comment:
Maybe it should be the whole spec? Just so that we don't have to be responsible for what info the decider uses to select which kernel to run?